File: Learn/Haskell/vector.cpp
#include <bits/stdc++.h>
using namespace std;

struct Vec {
  int x, y;
  void MakeVec(int a, int b) {
    x = a;
    y = b;
  }
  Vec operator + (const Vec k) {
    Vec a;
    a.x = x + k.x;
    a.y = y + k.y;
    return a;
  }
}v1, v2, v3;

int a, b, c, d, m, n;

int main() {
  cin >> a >> b >> c >> d;
  v1.MakeVec(a, b);
  v2.MakeVec(c, d);
  v3 = v1 + v2;
  cout << '[' << v3.x << ',' << v3.y << "]\n";
}

File: Learn/Haskell/vector.hs
data Vec = MakeVec Int Int
addVec :: Vec -> Vec -> Vec
addVec (MakeVec x1 y1) (MakeVec x2 y2) = 
  MakeVec (x1 + x2) (y1 + y2)

main :: IO()
main = do 
  l1 <- getLine
  let [a, b] = map (read::String->Int) (words l1)
  l2 <- getLine
  let [c, d] = map (read::String->Int) (words l2)
  let MakeVec m n = addVec (MakeVec a b) (MakeVec c d)
  print [m, n]
